src/levanter/main/train_asr.py

```diff
@@ -0,0 +1,199 @@
+import dataclasses
+import logging
+import os
+from dataclasses import dataclass, field
+from typing import Optional, Union
+
+import jax.random as jrandom
+
+import haliax as hax
+from haliax import Axis
+from haliax.partitioning import named_jit, round_axis_for_partitioning
+
+import levanter
+from levanter import callbacks
+from levanter.compat.hf_checkpoints import HFCompatConfig, save_hf_checkpoint_callback
+from levanter.data.audio import AudioIODatasetConfig, AudioTextDataset
+from levanter.models.asr_model import ASRConfig
+from levanter.models.whisper import WhisperConfig
+from levanter.optim import AdamConfig, OptimizerConfig
+from levanter.trainer import Trainer, TrainerConfig
+from levanter.utils.jax_utils import parameter_count
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class TrainASRConfig:
+    data: AudioIODatasetConfig = field(default_factory=AudioIODatasetConfig)
+    trainer: TrainerConfig = field(default_factory=TrainerConfig)
+    model: ASRConfig = field(default_factory=WhisperConfig)
+    optimizer: OptimizerConfig = field(default_factory=AdamConfig)
+    batch_size: int = 16
+
+    # config related to continued pretraining
+    initialize_from_hf: Union[bool, str] = False
+    """if provided, this will override the model config in the config. if true, use the default hf checkpoint for this model class"""
+    use_hf_model_config: bool = False  # if true, replace the model config with the hf config from the checkpoint
+
+    # TODO: atm we don't support loading from a checkpoint that has a different tokenizer. this is a bit annoying
+    # TODO: atm you have to at least specify a levanter model config with the same type as the hf checkpoint
+
+    hf_save_path: Optional[str] = None
+    hf_upload: Optional[str] = None
+    hf_save_steps: int = 10000
+
+    update_hessian_steps: int = 10
```

Comment:
copy pasta?

Reply:
Oops, yes. Don't know why mypy didn't yell at me for unused var.

Reply:
it's a memvar so I don't think it cares. it's probably vestigial in train_lm too